File: crates/facet_generate/src/generation/tests/with_enums_across_namespaces/snapshots/namespace_to_root/typescript_json/kv.ts
import * as $json from "./serde/json";
import * as Example from "./example";

export class Entry {
    constructor (public level: Example.Level, public outcome: Example.Outcome) {
    }

    static toJson(value: Entry): $json.JsonValue {
        return {
            "level": Example.toJsonLevel(value.level),
            "outcome": Example.toJsonOutcome(value.outcome),
        };
    }

    static fromJson(json: unknown): Entry {
        const obj = $json.readObject(json, "Entry");
        return new Entry(
            Example.fromJsonLevel($json.field(obj, "level")),
            Example.fromJsonOutcome($json.field(obj, "outcome")),
        );
    }

    static jsonSerialize(value: Entry): string {
        return $json.stringify(Entry.toJson(value));
    }

    static jsonDeserialize(text: string): Entry {
        return Entry.fromJson($json.parse(text));
    }
}

File: crates/facet_generate/src/generation/tests/with_enums_across_namespaces/snapshots/namespace_to_root/swift_json/Sources/Kv/Kv.swift
import Example
import Serde

public struct Entry: Hashable, Equatable, Codable {
    public var level: Example.Level
    public var outcome: Example.Outcome

    public init(level: Example.Level, outcome: Example.Outcome) {
        self.level = level
        self.outcome = outcome
    }

    enum CodingKeys: String, CodingKey {
        case level
        case outcome
    }

    public func jsonSerialize() throws -> [UInt8] {
        return try Serde.jsonSerialize(self)
    }

    public static func jsonDeserialize(input: [UInt8]) throws -> Entry {
        return try Serde.jsonDeserialize(Entry.self, from: input)
    }
}
